server/src-lib/Hasura/GraphQL/Resolve/Types.hs

```diff
@@ -277,14 +277,32 @@ Node id data:
 We are using JSON format for encoding and decoding the node id. The JSON
 schema looks like following
 
-'["<table-schema>", "<table-name>", "column-1", "column-2", ... "column-n"]'
+'[<version-integer>, "<table-schema>", "<table-name>", "column-1", "column-2", ... "column-n"]'
 
-It is represented in the type @'NodeIdData'.
+It is represented in the type @'NodeIdData'. The 'version-integer' represents the JSON
+schema version to enable any backward compatibility if it is broken in upcoming versions.
 
 The stringified JSON is Base64 encoded and sent to client. Also the same
 base64 encoded JSON string is accepted for 'node' field resolver's 'id' input.
 -}
 
+data NodeIdVersion
```

Comment:
We don't need to have a separate type for the version I think. Instead, something like this would model the id correctly?
```haskell
data NodeId
  = NodeIdV1 !V1NodeId
```